Add TypeScript declarations

favna opened the pull request:
I've recently picked up learning TypeScript and this lib didn't have a declarations file yet so yay here we go. 

I also updated the README usage examples part so you can port that over to boorujs.org. And I hope you don't mind I added myself to contributors =P

Lastly I bumped the version to 1.3.5 while I was editing anyway.

favna committed:
Add TypeScript declarations

favna committed:
Update README, bump version patch

AtoraSuunva commented on readme.md:
````diff
@@ -58,6 +59,40 @@ Booru.search(site, [tag1, tag2], {limit: 1, random: false})
 })
 ```
 
+**in TypeScript**
````
Adding "In TypeScript" seems odd, since TS is a superscript of JS and `booru` doesn't change much if you use it in TS versus JS (I'm fairly sure the JS example would work fine in JS, and the TS example doesn't add or change much)

favna replied:
addressed in a544bbb

AtoraSuunva commented on typings/booru.d.ts:
```diff
@@ -0,0 +1,107 @@
+declare namespace booru {
+    function commonfy(images: any): any;
```
`commonfy` should be removed from the typings, since it's going to be removed eventually & is already deprecated

favna replied:
addressed in a544bbb

favna committed:
Resolve typing issue

Resolves the TypeScript compiler throwing this error:

Argument of type 'string | number' is not assignable to parameter of type 'string'.
  Type 'number' is not assignable to type 'string'.

favna committed:
Update README and typings as requested

Favna commented:
@AtlasTheBot I'm so sorry I didn't get to this earlier. I truly completely forgot. I've resolved the requested changes and I've also quickly fixed a minor typings issue you had in Index (since you had @ts-check enabled, VSCode was finding the issue).

---

On a complete sidenote I was curious, what would you say to a complete rewrite of the lib into TypeScript? I've been doing a LOT of TypeScript work lately and I'd be willing to do that, but I need your green light since it will essentially change the code some what. It would be great if you could make that change to commonfy (and anything else pending) you said you would prior though as the compilation from TS to JS itself will generate the proper declaration files.

on a related note also coming back to what you said:

> I'm fairly sure the JS example would work fine in JS, and the TS example doesn't add or change much

It technically would but it is generally to be avoid to use commonJS require & module.export in TypeScript in favor of ES6 import & export. (not to mention that the compiler would trip on not defining types but that is beside the point).

You can check some of my recent TypeScript work [here (awesome-converter)](https://github.com/Favna/awesome-converter) and [here (awesome-querystring)](https://github.com/Favna/awesome-querystring)
